cloud/openstack/os_stack.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+# (c) 2016, Mathieu Bultel <mbultel@redhat.com>
+# (c) 2016, Steve Baker <sbaker@redhat.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+from time import sleep
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_stack
+short_description: Add/Remove Heat Stack
+extends_documentation_fragment: openstack
+version_added: "2.1"
+author: "Mathieu Bultel (matbu), Steve Baker (steveb)"
+description:
+   - Add or Remove a Stack to an OpenStack Heat
+options:
+    state:
+      description:
+        - Indicate desired state of the resource
+      choices: ['present', 'absent']
+      required: false
+      default: present
+    name:
+      description:
+        - Name of the stack that should be created
+      required: true
+      default: None
+    template:
+      description:
+        - Path of the template file to use for the stack creation
+      required: false
+      default: None
+    environment:
+      description:
+        - List of environment files that should be used for the stack creation
+      required: false
+      default: None
+    parameters:
+      description:
+        - Dictionary of parameters for the stack creation
+      required: false
+    rollback:
+      description:
+        - Rollback stack creation
+      required: false
+      default: false
+    timeout:
+      description:
+        - Maximum number of seconds to wait for the stack creation
+      required: false
+      default: 3600
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+EXAMPLES = '''
+---
+- name: create stack
+  ignore_errors: True
+  register: stack_create
+  os_heat:
+    name: "{{ stack_name }}"
+    state: present
+    template: "/path/to/my_stack.yaml"
+    environment:
+    - /path/to/resource-registry.yaml
+    - /path/to/environment.yaml
+    parameters:
+        os_user: {{ os_user }}
+        os_password: {{ os_password }}
+        os_tenant: {{ os_tenant }}
+        os_auth_url: {{ os_auth_url }}
+        bmc_flavor: m1.medium
+        bmc_image: CentOS
+        key_name: default
+        private_net: {{ private_net_param }}
+        node_count: 2
+        name: undercloud
+        image: CentOS
+        my_flavor: m1.large
+        external_net: {{ external_net_param }}
+'''
+
+RETURN = '''
+#
+'''
+
```

Comment:
It returns the result of get_stack, which includes an outputs dict that is likely useful for later tasks. Showing the get_stack structure would be useful here, and maybe an echo task which prints an output value?